examples/ethernet-rtic-nucleo-h723zg.rs

```diff
@@ -0,0 +1,235 @@
+//! Demo for Nucleo-H723ZG eval board using the Real Time for the Masses
+//! (RTIC) framework.
+//!
+//! This demo responds to pings on 192.168.1.99 (IP address hardcoded below)
+//!
+//! We use the SysTick timer to create a 1ms timebase for use with smoltcp.
+//!
+//! The ethernet ring buffers are placed in AXI SRAM, where they can be
+//! accessed by both the core and the Ethernet DMA.
+//!
+//! Run like
+//!
+//! `cargo flash --example ethernet-rtic-nucleo-h723zg --features=ethernet,stm32h735 --chip=STM32H723ZGTx`
+#![deny(warnings)]
+#![no_main]
+#![no_std]
+
+#[macro_use]
+#[allow(unused)]
+mod utilities;
+
+use core::sync::atomic::AtomicU32;
+
+use smoltcp::iface::{Config, Interface, SocketSet, SocketStorage};
+use smoltcp::time::Instant;
+use smoltcp::wire::{HardwareAddress, IpAddress, IpCidr};
+
+use stm32h7xx_hal::{ethernet, rcc::CoreClocks, stm32};
+
+/// Configure SYSTICK for 1ms timebase
+fn systick_init(mut syst: stm32::SYST, clocks: CoreClocks) {
+    let c_ck_mhz = clocks.c_ck().to_MHz();
+
+    let syst_calib = 0x3E8;
+
+    syst.set_clock_source(cortex_m::peripheral::syst::SystClkSource::Core);
+    syst.set_reload((syst_calib * c_ck_mhz) - 1);
+    syst.enable_interrupt();
+    syst.enable_counter();
+}
+
+/// TIME is an atomic u32 that counts milliseconds.
+static TIME: AtomicU32 = AtomicU32::new(0);
+
+/// Locally administered MAC address
+const MAC_ADDRESS: [u8; 6] = [0x02, 0x00, 0x11, 0x22, 0x33, 0x44];
+
+/// Ethernet descriptor rings are a global singleton
+#[link_section = ".axisram.eth"]
+static mut DES_RING: ethernet::DesRing<4, 4> = ethernet::DesRing::new();
+
+/// Net storage with static initialisation - another global singleton
+pub struct NetStorageStatic<'a> {
+    socket_storage: [SocketStorage<'a>; 8],
+}
+static mut STORE: NetStorageStatic = NetStorageStatic {
+    // Garbage
+    socket_storage: [SocketStorage::EMPTY; 8],
```

Comment:
This static variable will be initialized by the [cortex-m-rt ](https://docs.rs/cortex-m-rt/latest/cortex_m_rt/#features) runtime, so the comment `// Garbage` should be removed. In the past - or perhaps somewhere else - a `#[link_section]` directive was used to place this static variable somewhere else, and this did apply.

Reply:
Copy, removed

Reply:
Thanks!